frontend/src/APIClients/CourseAPIClient.ts

```diff
@@ -79,7 +137,7 @@ const updateCourse = async (
     "accessToken",
   )}`;
   try {
-    const { data } = await baseAPIClient.put(`/course/${id}`, course, {
+    const { data } = await baseAPIClient.put(`/course/${id}`, serializeUpdateCourseRequest(course), {
       headers: { Authorization: bearerToken },
     });
     return data;
```

Comment:
I might not be following correctly, but do we need to deserialize the course response here?

Reply:
Yes we do, good catch!